integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/mutation/CarbonProjectForUpdateCommand.scala

```diff
@@ -99,6 +96,8 @@ private[sql] case class CarbonProjectForUpdateCommand(
     val metadataLock = CarbonLockFactory
       .getCarbonLockObj(carbonTable.getAbsoluteTableIdentifier,
         LockUsage.METADATA_LOCK)
+    val compactionLock = CarbonLockFactory.getCarbonLockObj(carbonTable
+      .getAbsoluteTableIdentifier, LockUsage.COMPACTION_LOCK)
```

Comment:
Add update lock even for delete command as well

Reply:
okay